pkgs/tools/audio/vosk/kaldi.nix

```diff
@@ -0,0 +1,125 @@
+{ lib
+, openfst
+, fetchFromGitHub
+, stdenv
+, writeText
+, autoconf
+, automake
+, libtool
+, gfortran
+, openblas
+}:
+let
+  openfst' = openfst.overrideAttrs (oldAttrs: {
+    version = "1.8.0"; # this needs to match the version from kaldi/tools/Makefile
+
+    # not pinned, so it should be updated to the latest version when updating kaldi
+    src = fetchFromGitHub {
+      owner = "alphacep";
+      repo = "openfst";
+      rev = "7dfd808194105162f20084bb4d8e4ee4b65266d5";
+      sha256 = "sha256-XiPR4AaSa/7OqYoYZOwlW3UhAsYmscBE36xffI2gPPg=";
+    };
+
+    # from OPENFST_CONFIGURE in kaldi/tools/Makefile
+    configureFlags = [
+      "--enable-shared"
+      "--enable-far"
+      "--enable-ngram-fsts"
+      "--enable-lookahead-fsts"
+      "--with-pic"
+    ];
+  });
+
+  # pinned in kaldi/tools/Makefile
+  cub = fetchFromGitHub {
+    owner = "nvidia";
+    repo = "cub";
+    rev = "1.8.0";
+    sha256 = "sha256-j52BSkTItExznQApZbv868ipU4SCAu0EOqljzzKnIdk=";
+    meta.license = lib.licenses.bsd3;
+  };
+
+  # When updating, always update the git revision (rev) and the version base.
+  # The version base can be obtained by running src/base/get_version.sh in the kaldi source.
+  # Only the first three components (major.minor.patch) should be put in here,
+  # the last one (short git commit hash) will automatically be appended
+  version_base = "5.5.1046";
+  rev = "76cd51d44c0a61e3905c35cadb2ec5f54f3e42d1";
+in
+stdenv.mkDerivation rec {
```

Comment:
Alternately, we might want to override the already existing `kaldi` package (either `src` or `patches`) like in https://github.com/NixOS/nixpkgs/pull/185148/files#diff-a4997bb1585454092e3e7a29bb551970a156f4c7fb1135984c17303027f022e5R10-R32

That way we will not have unnecessary duplication of code that will inevitably get out of sync.